lib/main.go

```diff
@@ -251,30 +269,22 @@ func GetStringBodyHTTPRequest(r *http.Request) *string {
 	headers, _ := httputil.DumpRequest(r, false)
 	headersAndBody, _ := httputil.DumpRequest(r, true)
 	body := headersAndBody[len(headers):]
-	stringBody := string(body)
-
-	re := regexp.MustCompile(`(?s)(.*)`)
-	groups := re.FindStringSubmatch(stringBody)
-
-	if len(groups) > 0 {
-		// fmt.Printf("GetStringBodyHTTPRequest: printing request Body: %+v\n", groups[0])
-		return &groups[0]
-	}
-
-	// fmt.Printf("GetStringBodyHTTPRequest: no body to print on request Body\n")
-	return nil
+	s := strings.TrimSpace(string(body))
+	return &s
 }
 
 // GetStringBodyHTTPRequestJSON REQUIRE THEM TO DOCUMENT THIS FUNCTION
 func GetStringBodyHTTPRequestJSON(r *http.Request) *string {
 	result := GetStringBodyHTTPRequest(r)
 	if result != nil {
-		re := regexp.MustCompile(`({.*})`)
-		groups := re.FindStringSubmatch(*result)
-		if len(groups) > 0 {
-			return &groups[0]
+		rBytes := []byte(*result)
+		start := bytes.IndexByte(rBytes, byte('{'))
+		for i := len(rBytes) - 1; i > 0; i-- {
+			if string(rBytes[i]) == "}" {
```

Comment:
Pelo que entendi, se essa `var` for declarada a nível de pacote (`var closingCurlyBrackets = []byte("}")`), não será necessário fazer a conversão para `string`.

Reply:
@rodrigosdo porque comparar byte array é mais performático, certo?
Se sim, melhor eu comparar como `if rBytes[i] == []byte("}")` então porque ficar instanciando um monte de string que nem sei se vou usar também não sei se é performático.

Reply:
@rodrigosdo Sim, eu entendo a reutilização. Foi o caso daquelas regexps, a diferença é que eu sei que as regexps de SKU serão instancias reutilizadas (e como serão) já essa, talvez seja instanciada por um app que importe o package (alocando recurso) e nunca seja usada.
Vou levar levar elas pra uma instancia reutilizável mas vale essa reflexão de quando usar ou não esse approach.